tests/ion_schema_tests_runner.rs

```diff
@@ -0,0 +1,167 @@
+    "ion-schema-tests/ion_types/$list.isl",
+    "ion-schema-tests/ion_types/$lob.isl",
+    "ion-schema-tests/ion_types/$null.isl",
+    "ion-schema-tests/ion_types/$number.isl",
+    "ion-schema-tests/ion_types/$sexp.isl",
+    "ion-schema-tests/ion_types/$struct.isl",
+    "ion-schema-tests/ion_types/$symbol.isl",
+    "ion-schema-tests/ion_types/$text.isl",
+    "ion-schema-tests/ion_types/$timestamp.isl",
+];
+
+#[test]
+fn validation_tests() {
+    // create a schema system for validation test
+    let authority = FileSystemDocumentAuthority::new(Path::new(TEST_ROOT_DIR));
+    let mut schema_system = SchemaSystem::new(vec![Box::new(authority)]);
+
+    for path in SCHEMA_TEST_LIST {
+        print!("Reading {}... ", path);
+
+        // get the schema content from given schema file path
+        let ion_content = fs::read(path).unwrap();
+        let iterator = element_reader().iterate_over(&ion_content).unwrap();
+        let schema_content = iterator
+            .collect::<Result<Vec<OwnedElement>, IonError>>()
+            .unwrap();
+
+        let type_store = &mut TypeStore::new();
+        let mut invalid_values: Vec<OwnedElement> = vec![];
+        let mut valid_values: Vec<OwnedElement> = vec![];
+        let mut type_def: Option<TypeDefinitionImpl> = None;
+        let mut test_failed: bool = false;
+
+        for element in schema_content {
+            let annotations: Vec<&OwnedSymbolToken> = element.annotations().collect();
+            if annotations.contains(&&text_token("invalid_type")) {
+                // check for an invalid type validation
+                let type_def_result = schema_system.schema_type_from_element(&element, type_store);
+                if type_def_result.is_ok() {
+                    println!("ERROR: expected error for invalid type {:?}", element);
+                    test_failed = true;
+                    break;
+                }
+            } else if annotations.contains(&&text_token("invalid")) {
+                // get invalid values to validate
+                invalid_values = element
+                    .as_sequence()
+                    .unwrap()
+                    .iter()
+                    .map(|v| v.to_owned())
+                    .collect();
+            } else if annotations.contains(&&text_token("valid")) {
+                // get valid values to validate
+                valid_values = element
+                    .as_sequence()
+                    .unwrap()
+                    .iter()
+                    .map(|v| v.to_owned())
+                    .collect();
+            } else if annotations.contains(&&text_token("type")) {
+                // get type definition for type validation
+                type_def = Some(
+                    schema_system
+                        .schema_type_from_element(&element, type_store)
+                        .unwrap(),
+                );
+            } else {
+                continue;
+            }
```

Comment:
`else` block is redundant.
```suggestion
            }
```